packages/next/build/webpack/config/blocks/css/index.ts

```diff
@@ -11,80 +11,99 @@ import {
+      return transformer
+    }
 
-// @ts-ignore backwards compat
-postcss.vendor = {
-  /**
-   * Returns the vendor prefix extracted from an input string.
-   *
-   * @param {string} prop String with or without vendor prefix.
-   *
-   * @return {string} vendor prefix or empty string
-   *
-   * @example
-   * postcss.vendor.prefix('-moz-tab-size') //=> '-moz-'
-   * postcss.vendor.prefix('tab-size')      //=> ''
-   */
-  prefix: function prefix(prop: any) {
-    const match = prop.match(/^(-\w+-)/)
+    let cache: any
+    Object.defineProperty(creator, 'postcss', {
+      get() {
+        if (!cache) cache = creator()
+        return cache
+      },
+    })
 
-    if (match) {
-      return match[0]
+    creator.process = function (css: any, processOpts: any, pluginOpts: any) {
+      return postcss([creator(pluginOpts)]).process(css, processOpts)
     }
 
-    return ''
-  },
+    return creator
+  }
 
-  /**
-   * Returns the input string stripped of its vendor prefix.
-   *
-   * @param {string} prop String with or without vendor prefix.
-   *
-   * @return {string} String name without vendor prefixes.
-   *
-   * @example
-   * postcss.vendor.unprefixed('-moz-tab-size') //=> 'tab-size'
-   */
-  unprefixed: function unprefixed(prop: any) {
-    return prop.replace(/^-\w+-/, '')
-  },
-}
+  // @ts-ignore backwards compat
+  postcss.vendor = {
+    /**
+     * Returns the vendor prefix extracted from an input string.
+     *
+     * @param {string} prop String with or without vendor prefix.
+     *
+     * @return {string} vendor prefix or empty string
+     *
+     * @example
+     * postcss.vendor.prefix('-moz-tab-size') //=> '-moz-'
+     * postcss.vendor.prefix('tab-size')      //=> ''
+     */
+    prefix: function prefix(prop: any) {
+      const match = prop.match(/^(-\w+-)/)
 
-// RegExps for all Style Sheet variants
-export const regexLikeCss = /\.(css|scss|sass)$/
+      if (match) {
+        return match[0]
+      }
 
-// RegExps for Style Sheets
-const regexCssGlobal = /(?<!\.module)\.css$/
-const regexCssModules = /\.module\.css$/
+      return ''
+    },
 
-// RegExps for Syntactically Awesome Style Sheets
-const regexSassGlobal = /(?<!\.module)\.(scss|sass)$/
-const regexSassModules = /\.module\.(scss|sass)$/
+    /**
+     * Returns the input string stripped of its vendor prefix.
+     *
+     * @param {string} prop String with or without vendor prefix.
+     *
+     * @return {string} String name without vendor prefixes.
+     *
+     * @example
+     * postcss.vendor.unprefixed('-moz-tab-size') //=> 'tab-size'
+     */
+    unprefixed: function unprefixed(prop: any) {
+      return prop.replace(/^-\w+-/, '')
+    },
+  }
+
+  const postCssPlugins = await getPostCssPlugins(
+    ctx.rootDirectory,
+    ctx.supportedBrowsers,
+    !ctx.future.strictPostcssConfiguration
+  )
+
+  postcssInstance = {
+    postcss,
+    postcssWithPlugins: postcss(postCssPlugins),
+  }
+  return postcssInstance
```

Comment:
This kind of caching doesn't work here. This way there is a race condition and it's potentically loading postcss multiple times. You need to cache the Promise instead.

``` js
let postcssInstancePromise: Promise<any>
function lazyPostCSS(ctx: ConfigurationContext) {
  if (!postcssInstancePromise) {
    postcssInstancePromise = (async() {
      // async initialization code here
    })();
  }
  return postcssInstancePromise;
}
```

Reply:
Done!